pfSense-pkg-API/files/etc/inc/api/models/APIFirewallRuleCreate.inc

```diff
@@ -56,7 +56,10 @@ class APIFirewallRuleCreate extends APIModel {
     private function __validate_interface() {
         # Check for our required 'interface' payload value
         if (isset($this->initial_data['interface'])) {
-            $this->initial_data['interface'] = APITools\get_pfsense_if_id($this->initial_data['interface']);
+            # only check for if id if the interface name is not a valid group name
+            if (!APITools\is_ifgroup($this->initial_data['interface'])) {
```

Comment:
I understand and agree! The support for firewall rules on interface groups is now moved to the `APITools\get_pfsense_if_id`  function and can be used by an optional flag.

I have also created the Endpoint `APIInterfaceGroup` it supports all the request methods GET, POST, PUT and DELETE.
While doing so I spotted that pfSense supports deleting an interface group while firewall rules are still present. In this case, the rules are still viewable via the API, so they must be lurking around somewhere but not visible through the pfSense ui itself.
I have decided to not included any logic to avoid deleting interface groups with active firewall rules, as pfSense itself supports this behaviour.